CMakeLists.txt

```diff
@@ -318,13 +346,65 @@ elseif(DEBIAN_BASED)
   add_definitions(-DDEBIAN_BASED=1)
 endif()
 
-# Make sure the generated paths exist
-execute_process(COMMAND mkdir -p "${CMAKE_BINARY_DIR}/generated")
+if(WIN32)
+  set(MKDIR_OPTS "")
```

Comment:
If the default `OPTS` should be set to nothing, let's break it out of the conditional, then only set if !WINDOWS. Then if someone does some logic ordering in these conditionals here, there wont be a requirement (or confusion) about the OPTS.